### PR TITLE
save memory with cache headers

### PR DESCRIPTION
**Pull request by nwadams**
save a bunch of memory by not parsing all the cache headers up front
@lxdvs

### CODE REVIEW

File: src/com/android/volley/toolbox/DiskBasedCache.java [hunk 1]
```diff
@@ -440,7 +441,7 @@ public CacheHeader(String key, Entry entry) {
          * @param is The InputStream to read from.
          * @throws IOException
          */
-        public static CacheHeader readHeader(InputStream is) throws IOException {
+        public static CacheHeader readHeader(InputStream is, boolean readFull) throws IOException {
             CacheHeader entry = new CacheHeader();
 
             int magic = readInt(is);
```
Comment (elihart) [inline src/com/android/volley/toolbox/DiskBasedCache.java:450]: Unrelated to your changes... but man this is some obfuscated naming.

File: src/com/android/volley/toolbox/DiskBasedCache.java [hunk 2]
```diff
@@ -440,7 +441,7 @@ public CacheHeader(String key, Entry entry) {
          * @param is The InputStream to read from.
          * @throws IOException
          */
-        public static CacheHeader readHeader(InputStream is) throws IOException {
+        public static CacheHeader readHeader(InputStream is, boolean readFull) throws IOException {
```
Comment (elihart) [inline src/com/android/volley/toolbox/DiskBasedCache.java]: Could rename `readFull` to be more descriptive like `saveHeaders` or something.

**Comment by elihart** [comment]
Seems like a reasonable change as long as you're sure that not reading the headers fully will not manifest errors somewhere else or somewhere in the future.

**Comment by nwadams** [comment]
added fixup and fixed a minor bug.

**Comment by elihart** [comment]
Fixes look good to me

File: src/com/android/volley/toolbox/DiskBasedCache.java [hunk 3]
```diff
@@ -456,7 +460,13 @@ public static CacheHeader readHeader(InputStream is) throws IOException {
             entry.serverDate = readLong(is);
             entry.ttl = readLong(is);
             entry.softTtl = readLong(is);
-            entry.responseHeaders = readStringStringMap(is);
+
+            Map<String, String> headers = readStringStringMap(is);
```
Comment (lxdvs) [inline src/com/android/volley/toolbox/DiskBasedCache.java]: shouldn't this be moved into the if() clause for maximal memory savings?

Comment (nwadams) [inline src/com/android/volley/toolbox/DiskBasedCache.java]: ++ i'll make that optimization before i merge.

**Pull request merged**